### PR TITLE
add `stTransactionTest` testsuite

### PR DESCRIPTION
**Pull request by TAdev0**
Time spent on this PR: 15min

#446 

# Pull Request type

Please check the type of change your PR introduces:

-   [ ] Bugfix
-   [ ] Feature
-   [ ] Code style update (formatting, renaming)
-   [ ] Refactoring (no functional changes, no API changes)
-   [ ] Build-related changes
-   [ ] Documentation content changes
-   [x] Testing

# What is the new behavior?

# Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: blockchain-tests-skip.yml [hunk 1]
```diff
@@ -444,9 +444,11 @@ filename:
     - TestStoreGasPrices.json # ef-test #139
     - TestContractInteraction.json # ef-test #140
     - RecursiveCreateContractsCreate4Contracts.json # ef-test #141
+  stTransactionTest:
+    - HighGasPrice.json
```
Comment (khaeljy) [inline blockchain-tests-skip.yml:448]: This test fails due to a panic error when loading the file. 
I think the error occurred before the test case was executed, so skipping the file won't have any impact (to be confirmed).
We can therefore remove this file from the list of files to be skipped.

File: crates/ef-testing/tests/tests.rs [hunk 2]
```diff
@@ -123,7 +123,7 @@ mod blockchain_tests {
     // blockchain_tests!(st_static_flag_enabled, stStaticFlagEnabled); // 13 failing tests
     // blockchain_tests!(st_system_operations_test, stSystemOperationsTest); // 28 failing tests
     // blockchain_tests!(st_time_consuming, stTimeConsuming); // 14 failing tests
-    // blockchain_tests!(st_transaction_test, stTransactionTest); // failing due to: invalid length 62, expected a (both 0x-prefixed or not) hex string or byte array containing betwee
+    blockchain_tests!(st_transaction_test, stTransactionTest);
```
Comment (khaeljy) [inline crates/ef-testing/tests/tests.rs:126]: This test fails due to a panic error when loading the file.
I think it's better to leave this testsuite commented out, adding the number of the issue concerned.

Comment (TAdev0) [reply crates/ef-testing/tests/tests.rs:126]: @khaeljy  should i delete this PR then? if i remove the json file in the yml file and leave the testsuite commented out, then i have nothing to pull :)

Comment (khaeljy) [reply crates/ef-testing/tests/tests.rs:126]: I think you can prefix the comment with `ef-test #447 - ` to indicate that an issue is linked to this test.
_(This needs the approval of someone in the team)_

Comment (TAdev0) [reply crates/ef-testing/tests/tests.rs:126]: Good idea, let's do that. thank you.

**Comment by TAdev0** [comment]
@khaeljy  i made the modification. but now, there're other issues. github seems to compare to an old main

**Comment by TAdev0** [comment]
@khaeljy  should be good!

**Comment by khaeljy** [comment]
Please, can you run `cargo fmt --all`?

**Comment by TAdev0** [comment]
@danilowhk  ready to be merged

**Review by danilowhk — APPROVED** [review]

**Pull request merged**